test/ActionRequestTest.ts

```diff
@@ -247,12 +247,69 @@ describe("ActionRequestTest", function() {
         });
     });
 
+    describe("VirtualGoogleAssistant Tests Using Express", function() {
+        it("Calls the custom express with invalid parameters", async () => {
+            try {
+                const virtualGoogle = VirtualGoogleAssistant.Builder()
+                    .expressHandler("test.resources.expressProject.index", 3000)
+                    .handler("test.resources.expressProject.index")
+                    .directory("./test/resources/sampleProject")
+                    .create();
+            } catch (error) {
+                assert.equal(error.message, "Use only handler or expressHandler.");
+
+            }
+        });
+
+
+        it("Calls the custom express from a file", async () => {
+            const virtualGoogle = VirtualGoogleAssistant.Builder()
+                .expressHandler("test.resources.expressProject.index", 3000)
+                .directory("./test/resources/sampleProject")
+                .create();
+
+            const reply = await virtualGoogle.launch();
+
+            assert.equal(reply.speech, "Hello World");
+            assert.equal(reply.displayText, "Hello World Displayed");
+        });
+
+        it("Calls the custom express from a file twice with no port conflict", async () => {
+            const virtualGoogle = VirtualGoogleAssistant.Builder()
+                .expressHandler("test.resources.expressProject.index", 3000)
+                .directory("./test/resources/sampleProject")
+                .create();
+
+            const reply1 = await virtualGoogle.launch();
+            const reply2 = await virtualGoogle.launch();
+
+            assert.equal(reply1.speech, "Hello World");
+            assert.equal(reply2.displayText, "Hello World Displayed");
+        });
+
+        // With current implementation this case fails, since the port is occupied,
+        it.skip("Calls the custom express from a file twice with two instances in parallel", async () => {
+            const virtualGoogle1 = VirtualGoogleAssistant.Builder()
+                .expressHandler("test.resources.expressProject.index", 3000)
+                .directory("./test/resources/sampleProject")
+                .create();
+
+            const virtualGoogle2 = VirtualGoogleAssistant.Builder()
```

Comment:
How is this possible if they are both creating listeners?

Reply:
this test was based on the idea that in order of it supporting jest parallel testing it would need to be able to run more than one instance. 
With the current implementation it's not doable, which it's why it's skipped. I can just remove this test for now, but it's a test we can use if we change that part of the. implementation (for example listening on a specifc range of ports if we have a port in use exception)

Reply:
I didn't see the skip part - I get it now. I think we should leave it in for the reasons you say.